lib/db_connection.ex

```diff
@@ -1775,6 +1788,9 @@ defmodule DBConnection do
       {status, _conn_state} when status in [:idle, :transaction, :error] ->
         status_disconnect(conn, status, meter)
 
+      {:ok, query, result, _conn_status} ->
+        {:ok, {query, result}, meter}
```

Comment:
I packed them into a tuple because if we return `{:ok, query, result, meter}` here then we have to handle the following 4 return types:

- `{:ok, result, meter}`
- `{:ok, query, result, meter}`
- `{:ok, conn, result, meter}`
- `{:ok, conn, query, result, meter}` 

The first 2 are when you start a transaction inside of `run`. The other 2 when you do a fresh checkout. 

It seems less complicated to pack these into a tuple but please let me know what you think.